sim/battle-stream-example.ts

```diff
@@ -98,7 +98,7 @@ console.log("p2 is " + p2.constructor.name);
 	while ((chunk = await streams.omniscient.read())) {
 		console.log(chunk);
 	}
-})();
+})().catch(err => console.error(err));
```

Comment:
This seems unnecessary. Uncaught Promise rejections are already printed to console, right?

Reply:
In general, if a linter forces code that obviously isn't best practice, I prefer to use a linter-ignore comment, rather than do the non-best-practice thing.

Reply:
How uncaught promises are treated depends on the Node version, and at the moment an error message is logged to the console but in a different way than an uncaught exception (and this wasn't the behavior in the past). In this particular case, I would argue that the `catch` is in fact best practice, and adding a `// tslint:disable` would accomplish a similar thing in this case but would be less explicit.

That being said - I could care less about whether we `// tslint:disable` or `catch` here. This is not an important thing to get bogged down about in this PR.